### PR TITLE
Notion provider

### PR DESCRIPTION
**Pull request by mifopen**
Implemented https://notion.so oAuth provider

### CODE REVIEW

File: src/AspNet.Security.OAuth.Notion/NotionAuthenticationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            var context = new OAuthCreatingTicketContext(
+                principal,
+                properties,
+                Context,
+                Scheme,
+                Options,
+                Backchannel,
+                tokens,
+                tokens.Response.RootElement
+            );
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Notion/NotionAuthenticationHandler.cs]:
```suggestion
                tokens.Response.RootElement);
```

**Review by martincostello — COMMENTED** [review]
Just one minor style comment.

**Comment by mifopen** [comment]
@martincostello Could you give an estimation of how long it'll take to release the 5.0.5 version?

**Comment by martincostello** [comment]
There's 2 other PRs open to add new providers at the moment (#539, #543), so I'd like to release 5.0.5 with all three new providers once they're all ready to go.

If either of those take longer than the rest of the month to be ready to merge, then I'll look at releasing this one with the fix from #548 as 5.0.5 at the start of June.

**Comment by martincostello** [comment]
Provider now available from [NuGet.org](https://www.nuget.org/packages/AspNet.Security.OAuth.Notion/) - thanks again for your contribution.

**Pull request merged**